### PR TITLE
Adding before: command, giving heroku 60 seconds to load...

### PR DESCRIPTION
**Pull request by spriggsha**
...prior to the nightwatch tests kicking off.

### CODE REVIEW

File: tests/browser/tests/jssuccesstest.js [hunk 1]
```diff
@@ -6,6 +6,36 @@ var https = require('https');
...
 module.exports = {
 
+	before: function(browser, done){
+		var request = require('request');
+
+		console.log("Checking heroku app readiness...");
+		var i = 0;
+		getReadyOrNot();
+
+		function getReadyOrNot(){
```
Comment (commuterjoy) [inline tests/browser/tests/jssuccesstest.js]: perhaps call this `isHerokuReady`

File: tests/browser/tests/jssuccesstest.js [hunk 2]
```diff
@@ -6,6 +6,36 @@ var https = require('https');
...
+			i++;
+			request(browser.launch_url + '__gtg', function (error, response, body) {
+				if (!error && response.statusCode === 200) {
+					console.log("Response from " + browser.launch_url + "__gtg : " + response.statusCode);
+					console.log("Page is ready for testing");
+					done();
+				}else if(i >= 12){
+					console.log(browser.launch_url + " was not ready for testing after 60 seconds.  Aborting tests");
```
Comment (commuterjoy) [inline tests/browser/tests/jssuccesstest.js]: Does this fail without error? I think that's a good default for now while we build confidence in this system.

File: tests/browser/tests/jssuccesstest.js [hunk 3]
```diff
@@ -6,6 +6,36 @@ var https = require('https');
...
+		getReadyOrNot();
+
+		function getReadyOrNot(){
+			i++;
+			request(browser.launch_url + '__gtg', function (error, response, body) {
+				if (!error && response.statusCode === 200) {
+					console.log("Response from " + browser.launch_url + "__gtg : " + response.statusCode);
+					console.log("Page is ready for testing");
+					done();
+				}else if(i >= 12){
```
Comment (matthew-andrews) [inline tests/browser/tests/jssuccesstest.js]: does `before` itself not have a timeout?

File: tests/browser/tests/jssuccesstest.js (hunk 2 above)
Comment (spriggsha) [inline tests/browser/tests/jssuccesstest.js]: In the current implementation, if the page doesn't become ready, then yes, the test will run anyway, allowing us to grab some screenshots of what the application actually looked like in the browser.

File: tests/browser/tests/jssuccesstest.js (hunk 3 above)
Comment (spriggsha) [inline tests/browser/tests/jssuccesstest.js]: Nothing built-in, no.  http://nightwatchjs.org/guide#setup-teardown

File: tests/browser/tests/jssuccesstest.js (hunk 1 above)
Comment (spriggsha) [inline tests/browser/tests/jssuccesstest.js]: Fair enough.  Made the change.

**Comment by matthew-andrews** [comment]
@spriggsha we've made `next-build-tools deploy` block and wait for `/__gtg` is good.  is this still needed?

**Pull request closed without merging**